web/src/routes/RepositoryDetails/Tags/TagsList.tsx

```diff
@@ -121,6 +143,10 @@ export default function TagsList(props: TagsProps) {
       // This requires all tags to be present to build the signature map correctly
       const enrichedTags = enrichTagsWithCosignData(allTags);
       setTags(enrichedTags);
+
+      // Fetch pull statistics for all tags after cosign enrichment
+      await Promise.all(enrichedTags.map((tag: Tag) => getPullStats(tag)));
```

Comment:
We talked about this separately, but we don't want to load this for all tags up front because of the amount of requests on the frontend. This will either have to be a bulk API or called per tag row